eo-runtime/src/main/java/EOorg/EOeolang/EOtxt/EOregex$EOmatch.java

```diff
@@ -62,13 +64,28 @@ public class EOregex$EOmatch extends PhDefault {
                     final String r = new Param(regex, "r").strong(String.class);
                     final String txt = new Param(rho, "txt").strong(String.class);
                     final Matcher matcher = Pattern.compile(r).matcher(txt);
-                    if (matcher.matches()) {
-                        final Phi[] dest =
-                            new Phi[matcher.groupCount() == 0 ? 1 : matcher.groupCount()];
-                        return new Data.ToPhi(dest);
-                    } else {
-                        return new Data.ToPhi(new Phi[]{});
+                    List<Phi> dest = new ArrayList<>(0);
```

Comment:
@mximp dest should be `final`. Did you check this code with IntelliJ? For me it would complain in this case.

Reply:
Same as above